sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/GroupIntoBatches.java

```diff
@@ -221,6 +221,31 @@ private GroupIntoBatches(BatchingParams<InputT> params) {
             duration));
   }
 
+  public GroupIntoBatches<K, InputT> withByteSize(Long batchSizeBytes) {
+    checkArgument(
+        batchSizeBytes != null && batchSizeBytes < Long.MAX_VALUE && batchSizeBytes > 0,
+        "batchSizeBytes should be a non-negative value less than " + Long.MAX_VALUE);
+    return new GroupIntoBatches<>(
+        BatchingParams.create(
+            params.getBatchSize(),
+            batchSizeBytes,
+            params.getElementByteSize(),
+            params.getMaxBufferingDuration()));
+  }
+
+  public GroupIntoBatches<K, InputT> withByteSize(
```

Comment:
Please add javadoc, same for `Long` vs `long` as above.

Reply:
fixed 👍 thanks!

Reply:
cc: @reuvenlax since you seem to have made many of the recent changes to GroupIntoBatches.  I added these methods to allow for users to make use of multiple batching criteria simultaneously.  Another option for exposing this configuration option would be to implement a method like:

```java
public static <K, InputT> GroupIntoBatches<K, InputT> fromBatchingParams(BatchingParams params) {
...
}
```

Thoughts on the preferred approach?